### PR TITLE
Fix Bug in Flax-Speech-Encoder-Decoder Test

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
This PR fixes a bug in a Flax-Speech-Encoder-Decoder test that was failing after push (https://github.com/huggingface/transformers/runs/5493110105?check_suite_focus=true). Specifically, it amends the `test_freeze_feature_encoder` test to **omit** the random `decoder_attention_mask` from the input arguments of the speech-encoder-decoder model. This random `decoder_attention_mask` was resulting in `nan` values on the output logits of the `FlaxWav2Vec2BartModelTest`. Removing it as an input results in real valued output logits, and the test of concern passes following this change. The behaviour of the FlaxBartForCausalLM model to output `nan` values given a random `decoder_attention_mask` has been noted.

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]

File: tests/speech_encoder_decoder/test_modeling_flax_speech_encoder_decoder.py [hunk 1]
```diff
@@ -348,14 +344,14 @@ def compute_loss(
             feature_extractor_grads, feature_extractor_grads_frozen
         ):
             self.assertTrue((feature_extractor_grad_frozen == 0.0).all())
-            self.assert_difference(feature_extractor_grad, feature_extractor_grad_frozen, 1e-8)
+            self.assert_difference(feature_extractor_grad, feature_extractor_grad_frozen, 1e-10)
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply tests/speech_encoder_decoder/test_modeling_flax_speech_encoder_decoder.py:347]: Sure! I just set it in-tandem with the `assert_almost_equals` threshold that is run for the unfrozen gradients.
